tests/trainer/connectors/test_checkpoint_connector.py

```diff
@@ -0,0 +1,152 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+
+import torch
+
+from pytorch_lightning import Trainer
+from tests.helpers import BoringModel
+
+
+class HPCHookdedModel(BoringModel):
+
+    def __init__(self):
+        super().__init__()
+        self.hpc_save_called = 0
+        self.hpc_load_called = 0
+
+    def on_hpc_save(self, checkpoint):
+        assert "state_dict" in checkpoint
+        self.hpc_save_called += 1
+
+    def on_hpc_load(self, checkpoint):
+        assert "state_dict" in checkpoint
+        self.hpc_load_called += 1
+
+
+def test_hpc_hook_calls(tmpdir):
+    model = HPCHookdedModel()
+    trainer = Trainer(
+        default_root_dir=tmpdir,
+        max_steps=1,
+        checkpoint_callback=False,
+    )
+    trainer.fit(model)
+    connector = trainer.checkpoint_connector
+    connector.hpc_save(tmpdir, logger=trainer.logger)
+    assert model.hpc_save_called == 1
+    assert model.hpc_load_called == 0
+
+    # new training run, restore from hpc checkpoint file automatically
+    assert set(os.listdir(tmpdir)) == {"hpc_ckpt_1.ckpt", "lightning_logs"}
```

Comment:
set `logger=False` so `lightning_logs` isn't there?

Reply:
sure, I can give a mock to hpc_save instead :)